__test__/main.test.ts

```diff
@@ -185,5 +187,34 @@ describe("generate APT package names for ros_gz", () => {
 		await expect(
 			utils.generateROSAptPackageNames(["iron"], ["fortress", "garden"]),
 		).toEqual(["ros-iron-ros-gz", "ros-iron-ros-gzgarden"]);
+		await expect(
+			utils.generateROSAptPackageNames(["jazzy"], ["harmonic"]),
+		).toEqual(["ros-jazzy-ros-gz"]);
+		await expect(
+			utils.generateROSAptPackageNames(["rolling"], ["harmonic"]),
+		).toEqual(["ros-rolling-ros-gz"]);
+	});
+});
+
+describe("check for Gazebo vendor packages for ROS 2 distribution", () => {
```

Comment:
We could probably simplify this (by now) to just check the ROS 2 version without checking for Gazebo version. Logic as follow:
 * If ROS 2 is Jazzy or Rolling, use vendor packages
 * If ROS2 is not Jazzy or Rolling, use official/unofficial wrappers